pkg/destroy/azure/azure.go

```diff
@@ -116,7 +121,11 @@ func (o *ClusterUninstaller) Run() error {
 		waitCtx,
 		func(ctx context.Context) {
 			o.Logger.Debugf("deleting public records")
-			err = deletePublicRecords(ctx, o.zonesClient, o.recordsClient, o.privateZonesClient, o.privateRecordSetsClient, o.Logger, o.ResourceGroupName)
+			if o.Environment.Name == "AzureStackCloud" {
```

Comment:
Sorry I missed this earlier but instead of `"AzureStackClod"` you should use the constant, I believe it would be `azure.StackCloud`